### PR TITLE
Add a membership template and information

### PR DESCRIPTION
**Pull request by JoelMarcey**
This uses GitHub issue templates to create a form that will be filled out and filed as a GitHub issue in the repository for membership. 

This will be tested and tweaked once it gets merged since it has to integrate with the Issues page directly.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/membership.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        This distinction is mainly for housekeeping purposes. You can choose which type of membership you plan to be initially and change it as needed.
+
+        If you do become a member of the consortium, you will be added to the consortium mailing list, which is currently the only non-public asset of the consortium at this point.
+  - type: input
+    id: Name
+    attributes:
+      label: What is your name?
+    validations:
+      required: true
+  - type: dropdown
```
Comment (PLeVasseur) [inline .github/ISSUE_TEMPLATE/membership.yml:26]: Disclaimer: I haven't done a GitHub issue template before.

It seems like from the [docs](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/syntax-for-githubs-form-schema#dropdown) for the `dropdown` input type it's expected to have an `options` value in the `attributes`:

```
body:
- type: dropdown
  id: download
  attributes:
    label: How did you download the software?
    options:
      - Built from source
      - Homebrew
      - MacPorts
      - apt-get
    default: 0
  validations:
    required: true
```

I see there are options listed up here: Company, Institution, and so on, but they are not under an `options` value.

**Review by PLeVasseur — COMMENTED** [review]
I think it's reasonable to ask for the "why" of the interest. I think this makes the barrier low enough.

I'm leaving a comment for now, because I haven't done a GitHub issue template before and don't feel really qualified to review this. :sweat_smile: 

If the idea is to just merge it as-is and then you can experiment and adjust, then I can go ahead and approve.

**Comment by JoelMarcey** [comment]
Thanks for the feedback @PLeVasseur. I incorporated it - not sure what I was doing with the dropdown box (maybe that was an older way or something).

Yes, please approve and I will merge.

**Review by PLeVasseur — APPROVED** [review]
Approving so we can merge and check functionality :v:

**Review by ah801 — APPROVED** [review]

**Pull request merged**